beamer/agent/events.py

```diff
@@ -307,10 +314,19 @@ def fetch(self) -> list[Event]:
 
         result = []
         from_block = self._next_block_number
-        while from_block <= block_number:
-            to_block = min(block_number, BlockNumber(from_block + self._blocks_to_fetch))
+
+        if block_number - self._confirmation_blocks == 0:
+            return []
+
+        if block_number - self._confirmation_blocks < from_block:
+            return []
+
+        while from_block <= block_number - self._confirmation_blocks:
+            to_block = (
+                min(block_number, from_block + self._blocks_to_fetch) - self._confirmation_blocks
+            )
```

Comment:
I feel like the entire block above could be simplified by just subtracting the number of confirmation blocks from `block_number` after getting the block number from the RPC, i.e. after that `try...except` block above. Then you would not need to change anything here, right?